File: Python/Sorted_dict_to_obj.py
from types import SimpleNamespace


def sorted_dict_to_object(data: dict) -> SimpleNamespace:
    sorted_items = dict(sorted(data.items(), key=lambda item: item[0]))
    return SimpleNamespace(**sorted_items)


def main() -> None:
    user_data = {
        "age": 25,
        "name": "Daniel",
        "country": "USA",
        "score": 99,
    }

    obj = sorted_dict_to_object(user_data)

    print("Object:", obj)
    print("Name:", obj.name)
    print("Age:", obj.age)
    print("Country:", obj.country)
    print("Score:", obj.score)


if __name__ == "__main__":
    main()

File: Typescript/Sorted_dict_to_obj.ts
const sortedDictToObject = (data) => {
  const sortedEntries = Object.entries(data).sort(([a], [b]) =>
    a.localeCompare(b),
  );

  return Object.fromEntries(sortedEntries);
};

const main = () => {
  const userData = {
    age: 25,
    name: 'Daniel',
    country: 'USA',
    score: 99,
  };

  const obj = sortedDictToObject(userData);

  console.log('Object:', obj);
  console.log('Name:', obj.name);
  console.log('Age:', obj.age);
  console.log('Country:', obj.country);
  console.log('Score:', obj.score);
};

main();
